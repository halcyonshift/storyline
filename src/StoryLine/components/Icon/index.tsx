import {
    AccessTimeOutlined as AccessTime,
    AddOutlined as Add,
    AddCircleOutlined as AddCircle,
    AddPhotoAlternateOutlined as AddPhotoAlternate,
    AddLocationAltOutlined as AddLocationAlt,
    ArrowBackOutlined as ArrowBack,
    ArrowBackIosNewOutlined as ArrowBackIosNew,
    ArrowDownwardOutlined as ArrowDownward,
    ArrowForwardIosOutlined as ArrowForwardIos,
    ArrowUpwardOutlined as ArrowUpward,
    ArticleOutlined as Article,
    BookOutlined as Book,
    BugReportOutlined as BugReport,
    CasinoOutlined as Casino,
    CategoryOutlined as Category,
    ChangeCircleOutlined as ChangeCircle,
    ChatBubbleOutline as ChatBubble,
    ClearOutlined as Clear,
    CloseOutlined as Close,
    CodeOutlined as Code,
    DashboardCustomizeOutlined as DashboardCustomize,
    DeleteOutlined as Delete,
    DescriptionOutlined as Description,
    EditOutlined as Edit,
    ExpandMoreOutlined as ExpandMore,
    FileOpenOutlined as FileOpen,
    FilterListOutlined as FilterList,
    FormatAlignCenterOutlined as FormatAlignCenter,
    FormatAlignJustifyOutlined as FormatAlignJustify,
    FormatAlignLeftOutlined as FormatAlignLeft,
    FormatAlignRightOutlined as FormatAlignRight,
    FormatBoldOutlined as FormatBold,
    FormatItalicOutlined as FormatItalic,
    FormatListBulletedOutlined as FormatListBulleted,
    FormatListNumberedOutlined as FormatListNumbered,
    FormatStrikethroughOutlined as FormatStrikethrough,
    FormatUnderlinedOutlined as FormatUnderlined,
    GitHub,
    GroupOutlined as Group,
    GroupAddOutlined as GroupAdd,
    HistoryOutlined as History,
    HtmlOutlined as Html,
    HubOutlined as Hub,
    ImportExportOutlined as ImportExport,
    InfoOutlined as Info,
    InsightsOutlined as Insights,
    LabelImportantOutlined as LabelImportant,
    LibraryBooksOutlined as LibraryBooks,
    LiveHelpOutlined as LiveHelp,
    LocalOfferOutlined as LocalOffer,
    LocationOnOutlined as LocationOn,
    MenuBookOutlined as MenuBook,
    PeopleAltOutlined as PeopleAlt,
    Person4Outlined as Person4,
    PersonAddOutlined as PersonAdd,
    PersonAddAltOutlined as PersonAddAlt,
    PersonAddAlt1Outlined as PersonAddAlt1,
    PersonOutlined as Person,
    PictureAsPdfOutlined as PictureAsPdf,
    PostAddOutlined as PostAdd,
    PublishOutlined as Publish,
    QueueOutlined as Queue,
    RedoOutlined as Redo,
    RestorePageOutlined as RestorePage,
    SaveOutlined as Save,
    SearchOutlined as Search,
    SettingsOutlined as Settings,
    SettingsBackupRestoreOutlined as SettingsBackupRestore,
    StickyNote2Outlined as StickyNote2,
    SubjectOutlined as Subject,
    UndoOutlined as Undo,
    UpgradeOutlined as Upgrade,
    VisibilityOutlined as Visibility,
    VisibilityOffOutlined as VisibilityOff
} from '@mui/icons-material'

const components = {
    AccessTime,
    Add,
    AddCircle,
    AddPhotoAlternate,
    AddLocationAlt,
    ArrowBack,
    ArrowBackIosNew,
    ArrowDownward,
    ArrowForwardIos,
    ArrowUpward,
    Article,
    Book,
    BugReport,
    Casino,
    Category,
    ChangeCircle,
    ChatBubble,
    Clear,
    Close,
    Code,
    DashboardCustomize,
    Delete,
    Description,
    Edit,
    ExpandMore,
    FileOpen,
    FilterList,
    FormatAlignCenter,
    FormatAlignJustify,
    FormatAlignLeft,
    FormatAlignRight,
    FormatBold,
    FormatItalic,
    FormatListBulleted,
    FormatListNumbered,
    FormatStrikethrough,
    FormatUnderlined,
    GitHub,
    Group,
    GroupAdd,
    History,
    Html,
    Hub,
    ImportExport,
    Info,
    Insights,
    LabelImportant,
    LibraryBooks,
    LiveHelp,
    LocationOn,
    LocalOffer,
    MenuBook,
    PeopleAlt,
    Person4,
    PersonAdd,
    PersonAddAlt,
    PersonAddAlt1,
    Person,
    PictureAsPdf,
    PostAdd,
    Publish,
    Queue,
    Redo,
    RestorePage,
    Save,
    Search,
    Settings,
    SettingsBackupRestore,
    StickyNote2,
    Subject,
    Undo,
    Upgrade,
    Visibility,
    VisibilityOff
}

const Icon = ({ name }: { name: string }) => {
    const MuiIcon = components[name as keyof typeof components]
    return <MuiIcon aria-label={name} fontSize='inherit' color='inherit' />
}

export default Icon
