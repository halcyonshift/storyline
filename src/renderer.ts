import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
