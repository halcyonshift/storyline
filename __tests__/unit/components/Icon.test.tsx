import renderer from 'react-test-renderer'
import Icon from '@sl/components/Icon'
import { render, screen } from '../test-utils'

describe('<Icon />', () => {
    it('Shows icon', () => {
        render(<Icon name='Add' />)
        expect(screen.getByLabelText('Add')).toBeTruthy()
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Icon name='Add' />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
