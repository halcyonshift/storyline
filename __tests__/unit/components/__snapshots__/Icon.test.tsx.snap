// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<svg
  aria-hidden={true}
  aria-label="Add"
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
  data-testid="AddOutlinedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
  />
</svg>
`;
